pkg/config/multizone/multicluster.go

```diff
@@ -45,14 +45,20 @@ func DefaultGlobalConfig() *GlobalConfig {
 	}
 }
 
+type MetadataKeyValue struct {
+	Key   string
+	Value string
+}
+
 // Zone configuration
 type ZoneConfig struct {
 	// Kuma Zone name used to mark the zone dataplane resources
 	Name string `json:"name,omitempty" envconfig:"kuma_multizone_zone_name"`
 	// GlobalAddress URL of Global Kuma CP
 	GlobalAddress string `json:"globalAddress,omitempty" envconfig:"kuma_multizone_zone_global_address"`
 	// KDS Configuration
-	KDS *KdsClientConfig `json:"kds,omitempty"`
+	KDS                *KdsClientConfig   `json:"kds,omitempty"`
+	AdditionalMetadata []MetadataKeyValue `json:"additionalMetadata"`
```

Comment:
Should this be in the KdsClientConfig?

Reply:
To me it looks like an object passed to `NewKDSStream` instead of the current `cpConfig` I believe this should take a `func(ctx) *structpb.Struct` or equivalent

Reply:
Actually should this just be in the ZoneClientContext `pkg/kds/context` ?

Reply:
done